encoding/resampling.md

```diff
@@ -1,35 +1,429 @@
+consider this graph:
+
+![The sRGB gamma curve. x is the encoded value, y is the brightness.](images/gamma.png)
+
+When downscaling,
+adjacent pixels are essentially
+averaged together.
+Let’s say we want to merge
+pixels A and B.
+Averaging the gamma-encoded values
+would result in point C.
+However,
+C is clearly closer to A on the brightness scale.
+This means,
+*the average of the gamma-encoded values is always lower than the average brightness*,
+dimming the image slightly overall.
+The correct value,
+located at the average brightness,
+is marked as D on the graph.
+
+This dimming effect is usually only noticeable
+in dense color patterns,
+e.g. small black text on a white background,
+stars in the night sky, etc,
+and much less so in blurrier areas.
+
+See [this comparison][comp] for an example of linear vs gamma downscaling.
+
+However,
+this doesn’t mean downscaling in linear light
+is always the right choice,
+since it noticeably accentuates dark halos
+introduced by scaling.
+Besides,
+the dimming may even be desirable in some cases like black text on white paper,
+because it preserves readability.
+
+If you choose to downscale in linear light,
+make sure to use a sufficiently high bitdepth
+so as to not introduce banding.
+
+Example code for resizing in linear RGB light:
+
+```py
+linear = core.resize.Bicubic(src, format=vs.RGBS, transfer_in_s="709", transfer_s="linear", matrix_in_s="709")
+scaled_linear = core.resize.Bicubic(linear, 640, 360)
+scaled_gamma = core.resize.Bicubic(scaled_linear, format=src.format, transfer_s="709", transfer_in_s="linear", matrix_s="709")
+```
+
+[gamma]: https://en.wikipedia.org/wiki/Gamma_correction
+[comp]: https://slowpics.org/comparison/8103b2d1-b9d4-4d7e-b7a7-3197ae999244
+
+##### Upscaling in sigmoidized light
+
+In order to attenuate both dark and white halos 
+introduced by upscaling,
+you can resize through a sigmoidized colorspace.
+
+This means converting the linear RGB version of an image
+to a custom colorspace with an S-shaped intensitiy curve
+before scaling and converting it back afterwards.
+What this does, essentially,
+is decrease the image’s contrast
+by pushing extreme values of both dark and bright
+towards the middle.
+
+Quoting Nicholas Robidoux from ImageMagick:
+
+“You may decrease halos and increase perceptual sharpness by increasing the sigmoidal contrast (up to 11.5, say). Higher contrasts are especially recommended with greyscale images (even "false RGB greyscale" that have three proportional color channels). The downside of sigmoidization is that it sometimes produces "color bleed" artefacts that look a bit like cheap flexographic ("gummidruck") printing or chromatic aberration. In addition, sigmoidization's "fattening" of extreme light and dark values may not work for your image content. If such artefacts are obvious, push the contrast value down from 7.5 (to 5, for example, or even lower). Setting the contrast to 0 is equivalent to enlarging through linear RGB.“
```

Comment:
Use block quotes (`> `).

Also while you may not need to apply semantic line feeds since a quote is not to be changed, this line is excessively long, so it should at least be wrapped.

A source would also be nice.